sklearn/metrics/cluster/_unsupervised.py

```diff
@@ -338,7 +347,16 @@ def davies_bouldin_score(X, labels):
     labels = le.fit_transform(labels)
     n_samples, _ = X.shape
     n_labels = len(le.classes_)
-    check_number_of_labels(n_labels, n_samples)
+
+    # don't raise an error when le.classes_ = 1 or cle.classes_ = n_samples
+    # instead return the worst score = np.inf
```

Comment:
I think the comment to here is sufficient and the rest can be removed

Reply:
Done!